capa/features/extractors/dnfile/helpers.py

```diff
@@ -380,10 +381,11 @@ def resolve_nested_typeref_name(
 
         while isinstance(table_row.ResolutionScope.table, dnfile.mdtable.TypeRef):
             # Iterate through the typeref table to resolve the nested name
+            typeref_tb.append(table_row)
             typeref_name.append(name)
             name = table_row.TypeName
             table_row = get_dotnet_table_row(pe, dnfile.mdtable.TypeRef.number, table_row.ResolutionScope.row_index)
-            if table_row is None:
+            if table_row is None or table_row in typeref_tb:
```

Comment:
We use similar logic in `resolve_nested_typedef_name`. What are your thoughts on adding this check there?